### ISSUE TITLE
False positive on comments in macros

### ISSUE THREAD

**Issue by ekohl** [rpm-software-management/rpmlint#183]
In our spec we have the following line which triggers two warnings:
```bash
sed -i '/^---/ a #replace default location of "settings.d"\n:settings_directory: %{_sysconfdir}/%{name}/settings.d\n' \
        %{_sysconfdir}/%{name}/settings.yml
```

```
$ rpmlint packages/foreman/foreman-proxy/foreman-proxy.spec 
packages/foreman/foreman-proxy/foreman-proxy.spec:74: W: macro-in-comment %{_sysconfdir}
packages/foreman/foreman-proxy/foreman-proxy.spec:74: W: macro-in-comment %{name}
```

The exact line is at https://github.com/theforeman/foreman-packaging/blob/902e40846b68ba4f4fea20621ed51e858d1210c9/packages/foreman/foreman-proxy/foreman-proxy.spec#L74-L75

The problematic code is at https://github.com/rpm-software-management/rpmlint/blob/a87b6a9083ec26b31cf0f6018e2d1324aad5ead2/rpmlint/checks/SpecCheck.py#L478-L485
